### ISSUE TITLE
DeltaLake Table Source offline store implementation

### ISSUE THREAD

**Issue by dmatrix** [feast-dev/feast#1533]
**Is your feature request related to a problem? Please describe.**
Scalable data stores—Data warehouse and modern Data Lakes—are common today for offline store implementations or as a source of data ingestion. These stores are optimized and support ACID functionality where clean data can reside, and new data constantly can be updated and merged, accommodating data changes and observing schema evolutions. This ability to ingest data from or provide an offline store as a `feast.data_source.FileSource` from these stores extends
Feast's ecosystem to modern data lakes such as [Delta Lake](https://delta.io/), [Apache Hudi](https://hudi.apache.org/), or [Apache Iceberg](https://iceberg.apache.org/).  

A similar feature request for [HudiTableSource](https://github.com/feast-dev/feast/issues/1494) has been filed by @blvp

**Describe the solution you'd like**
Extend `feast.data_source.FileSource(...)` to take table names and locations to read from, for both local or remote sources.

**Describe alternatives you've considered**
I would have to save my [Delta Lake tables ](https://github.com/dmatrix/mlflow-tests/blob/master/py/delta-lake/create_weather_forecast_features.py)as a single parquet file and use that as `FileSource`, which may defeat the purpose of being able to ingest point-in-time data from these modern data lake sources.

**Comment by woop** [comment]
Just swinging back here @dmatrix. We'd love to support the sources you've laid out above. I think the biggest challenge from our side is just how many storage implementations we should support out of the box, and how many should be community contributed. We're trying to strike a balance here. Efforts like increased pluggability will be our short term solution, but I'd love to figure out if there are low hanging fruit that we haven't identified.

**Comment by dmatrix** [comment]
@woop Yes, it's a matter of striking a balance and compromise: go with the data source out of the box that's has wider adoption, reliable vendor support, or large contributing community. To some extent, one cannot discount the growing adoption of modern data lakes, based on the above open file formats. 

As part of an easily extensible Feast ecosystem, the idea of making data sources as _pluggable architecture_ is viable and attractive short term solution. Perhaps, we can solicit community help by labeling this issues as "Community Help Needed."

**Comment by Data-drone** [comment]
Happy to take a crack at this. I have had a little dive into the code source.

I can see that there is a `redshift` and `redshift_source` file and same with bigquery. Would I just need to reimplement that?

**Comment by creativedutchmen** [comment]
Looking at the source code for the Spark offline store, support for delta should be as easy as listing it in the `SparkSourceFormat`. Reading the table is done (as it should) with `df = spark_session.read.format(self.file_format).load(self.path)`. Reading a delta table is a matter of having `.format("delta")` and Spark will take care of the rest.

**Comment by franciscojavierarceo** [comment]
@HaoXuAI i think this was added, right?
